AutoEncoderTraining.py

```diff
@@ -0,0 +1,970 @@
+    plt.show()
+
+"""
+Funktion normalisiert einen Pixelwert
+"""
+def normalize(v):
+    return v / 255
+
+
+def reshape(dataset):
+    nData = []
+    for a in dataset:
+        b = np.delete(a,0,0)
+        nData.append(np.delete(b,0,0))
+    return asarray(nData)
+
+"""
+Grundstruktur für den Trainingsplot/Validierungsplot
+"""
+class VisdomLinePlotter(object):
+    """Plots to Visdom"""
+    def __init__(self, env_name='main'):
+        self.viz = Visdom()
+        self.env = env_name
+        self.plots = {}
+    def plot(self, var_name, split_name, title_name, x, y, ):
+        if var_name not in self.plots:
+            self.plots[var_name] = self.viz.line(X=np.array([x,x]), Y=np.array([y,y]), env=self.env, opts=dict(
+                legend=[split_name],
+                title=title_name,
+                xlabel='epochs',
+                ylabel=var_name
+            ))
+        else:
+            self.viz.line(X=np.array([x]), Y=np.array([y]), env=self.env, win=self.plots[var_name], name=split_name, update = 'append')
+
+
+"""
+Funktion trainiert einen Auto-Encoder mit RGB-Datensatz.
+Der Name des Datensatzes muss übergeben werden.
+Die Auto-Encoder-Architektur muss einmalig manuell im Code geändert werden.
+Trainierter Auto-Encoder wird am Ende der Funktion gespeichert (Pfad muss manuell angepasst werden).
+"""
+def startAutoEncoder(dataset):
+    global myDataYTrain
+    global myDataXTrain
+    global myDataYTest
+    global myDataXTest
+
+    #Überprüfung welcher Datensatz als Grundlage verwendet werden soll, anschließendes Laden des Datensatzes
+    if (dataset == 'SmallDataset_SpaceInvaders'):
+        split = True
+        grey = False
+        datasetTrain = load_dataset(
+            '/home/annika/BA-Datensaetze/SmallDataset_SpaceInvaders/smallDatasetTraining1_SpaceInvaders.npy')
+        datasetTrain2 = load_dataset(
+            '/home/annika/BA-Datensaetze/SmallDataset_SpaceInvaders/smallDatasetTraining2_SpaceInvaders.npy')
+        datasetTest = load_dataset(
+            '/home/annika/BA-Datensaetze/SmallDataset_SpaceInvaders/smallDatasetTest_SpaceInvaders.npy')
+        title = 'Auto-Encoder mit SmallDataset_SpaceInvaders'
+    elif(dataset == 'SmallDataset_Q*Bert_Mixed'):
+        split = False
+        grey = False
+        datasetTrain = load_dataset('/home/annika/BA-Datensaetze/SmallDataset_Q*Bert_Mixed/smallDatasetTraining.npy')
+        datasetTest = load_dataset('/home/annika/BA-Datensaetze/SmallDataset_Q*Bert_Mixed/smallDatasetTest.npy')
+        title = 'Auto-Encoder mit SmallDataset_Q*Bert_Mixed'
+    elif(dataset == 'SmallDataset_JustQbert'):
+        split = False
+        grey = False
+        datasetTrain = load_dataset('/home/annika/BA-Datensaetze/SmallDataset_JustQBert/smallDatasetTraining_JustQBert.npy')
+        datasetTest = load_dataset('/home/annika/BA-Datensaetze/SmallDataset_Q*Bert_Mixed/smallDatasetTest.npy')
+        title = 'Auto-Encoder mit SmallDataset_Q*Bert_JustQBert'
+
```

Comment:
Damit es später keinen Error gibt weil `split` und `grey` nicht definiert sind wäre ein Error gut wenn der Datensatz nicht übereinstimmt.

```suggestion
    else:
        raise RuntimeError("Not supported Dataset '{}' specified, aborting.".format(dataset))
```